Add Signup Modal

JPEscobari opened the pull request:
This update adds a new signup modal to the landing page, allowing users to create an account without leaving the page.

I've added a signup modal component, updating the landing page to conditionally render the signup modal, imitating what @agrimes23 did with the login modal.

* [`client/src/LandingPages/components/LandingPageSignupModal/SignupModal.jsx`](diffhunk://#diff-655bf276feb3d51ccde5a2d173ef3c748ef10ef51b24f6d46a30e44b1083cb2dR1-R175): Added a new `SignupModal` component that includes form fields for user signup, validation logic, and submission handling.
* [`client/src/LandingPages/components/LandingPageSignupModal/SignupModal.scss`](diffhunk://#diff-64a02363e4d59de3076fe0c4829fb7613c44ac245586f2d5e27b482cdce33c99R1-R109): Style page for the `SignupModal` component.

Landing Page Updates:

* [`client/src/LandingPages/LandingPage/LandingPage.jsx`](diffhunk://#diff-6d779c7a545d510fe66531ea3c6b788d1f84ce10bdb3dc9fb0128c557a0cc639R15-R20): Updated the landing page to import the new `SignupModal` component, added state management for showing the signup modal, and updated the signup button to trigger the modal instead of navigating to a separate signup page. [[1]](diffhunk://#diff-6d779c7a545d510fe66531ea3c6b788d1f84ce10bdb3dc9fb0128c557a0cc639R15-R20) [[2]](diffhunk://#diff-6d779c7a545d510fe66531ea3c6b788d1f84ce10bdb3dc9fb0128c557a0cc639L80-R90)

psantos2107 reviewed (changes requested):
Hey JP! Great job with this one! 
Just a couple of things:
1. Can you make the signup modal a bit wider? If you compare it to the wireframe, the width of your modal is much narrower. 
2. Can you change the default color of "already have an account? log in" to black? Right now it's blue by default. 

Thanks a bunch! These changes should be pretty quick.

JPEscobari commented:
Great observations [Paul Santos](https://huly.app/browse/?workspace=w-abigaildawso-knownative-663281b1-87be16d699-2a6bcd&_class=contact%3Aclass%3APerson&_id=669ad7769606cdeeaaeb77a6&label=Paul%20Santos).

I updated my pull request to match the wireframe:

* Fixed padding for the form in SignupModal.
* Adjusted the font size for signup-page__title.
* Changed the signup-page__signup-link color to black.

Also, `SignupModal.jsx`  now points to `SignupPage.scss` for styles.
